vault/plugin_reload.go

```diff
@@ -0,0 +1,103 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+// reloadPluginMounts reloads provided mounts, regardless of
+// plugin name, as long as the backend type is of plugin.
+func (c *Core) reloadMatchingPluginMounts(mounts []string) error {
+	c.mountsLock.Lock()
+	defer c.mountsLock.Unlock()
+
+	for _, mount := range mounts {
```

Comment:
Maybe it should use multierror to return which mounts were unable to be reloaded. That way it gets through as many as it can